lib/iris/analysis/trajectory.py

```diff
@@ -228,30 +240,32 @@ def interpolate(cube, sample_points, method=None):
     for coord, values in sample_points:
         if coord.ndim > 1:
             if method == "linear":
-                raise iris.exceptions.CoordinateMultiDimError("Cannot currently perform linear interpolation for multi-dimensional coordinates.")
+                raise iris.exceptions.CoordinateMultiDimError(
+                        "Cannot currently perform linear interpolation for "
+                        "multi-dimensional coordinates.")
             method = "nearest"
             break
 
-    # Use a cache with _nearest_neighbour_indices_ndcoords()
-    cache = {}
+    # Cache the interpolator
+    coords, points = zip(*sample_points)
+    if method in ["linear", None]:
+        scheme = iris.analysis.Linear()
+    elif method == "nearest":
+        scheme = iris.analysis.Nearest()
+    interpolator = scheme.interpolator(cube, coords)
 
     for i in range(trajectory_size):
-        point = [(coord, values[i]) for coord, values in sample_points]
-
-        if method in ["linear", None]:
-            column = iris.analysis.interpolate.linear(cube, point)
-            new_cube.data[..., i] = column.data
-        elif method == "nearest":
-            column_index = iris.analysis.interpolate._nearest_neighbour_indices_ndcoords(cube, point, cache=cache)
-            column = cube[column_index]
```

Comment:
In changing the nearest neighbour scheme 'iris.analysis.interpolate._nearest_neighbour_indices_ndcoords' to 'iris.analysis.Nearest().interpolator', the method of finding nearest points has changed significantly.

'_nearest_neighbour_indices_ndcoords' was a more physical result (used spherical-cartesian coordinate transformations to find the nearest point in 3-dimensional space) but was extremely long-winded, which used up a lot of processing time for each point.

This method is not used in any other functions, so has now become redundant, but may still be useful.

There are options regarding how to proceed with this:
1. Delete the function now that it is not used anywhere
2. Add a third option to trajectory (other than linear or nearest) to allow users to choose the old routine
3. Make a completely separate function to call this routine.

Personally I think it seems a shame to waste it, but differences in real-world results may be so negligible that there is no need to keep it.  I would like to know what people think about this.